basic-examples/bind-constants/workflow.yaml

```diff
@@ -0,0 +1,43 @@
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        repeated_inputs:
+          type:
+            type_id: ref
+            id: RepeatedValues
+        iterations:
+          type:
+            type_id: list
+            items:
+              id: SubRootObject
+              type_id: ref
+    RepeatedValues:
+      id: RepeatedValues
+      properties:
+        name:
+          type:
+            type_id: string
+        ratio:
+          type:
+            type_id: float
+    SubRootObject:
+      id: SubRootObject
+      properties:
+        loop_id:
+          type:
+            type_id: integer            
+            
+steps:
+  foreach_loop:
+    kind: foreach
+    items: !expr 'bindConstants($.input.iterations, $.input.repeated_inputs)'
+    workflow: subworkflow.yaml
+    parallelism: 1

```

Comment:
I'm on the fence about where and how much annotation to add.

This is a nice, compact file, so perhaps we only need to have a single block comment.  We could put it at the top of the file, or between `properties` and `repeated_inputs` in the input schema definition, or right here before `steps`.

The comment should basically say that the plugin will be called once for each entry in `$.input.iterations` and it will be passed an object which contains two fields, one containing the repeated input data and the other containing the appropriate entry from the iterations list.  (There's not much more to say...at the root, it's a fairly straightforward thing.)

Reply:
added the comments in the `foreach_loop` step.